docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -0,0 +1,372 @@
+* xref:#IERC20-approve[`++approve(spender: ContractAddress, amount: u256)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#IERC20-Transfer[`++Transfer(from: ContractAddress, to: ContractAddress, value: u256)++`]
+* xref:#IERC20-Approval[`++Approval(owner: ContractAddress, spender: ContratAddress, value: u256)++`]
+--
+
+[.contract-item]
+[[IERC20-total_supply]]
+==== `[.contract-item-name]#++total_supply++#++() → u256++` [.item-kind]#external#
+
+Returns the amount of tokens in existence.
+
+[.contract-item]
+[[IERC20-balance_of]]
+==== `[.contract-item-name]#++balance_of++#++(account: ContractAddress) → u256++` [.item-kind]#external#
+
+Returns the amount of tokens owned by `account`.
+
+[.contract-item]
+[[IERC20-allowance]]
+==== `[.contract-item-name]#++allowance++#++(owner: ContractAddress, spender: ContractAddress) → u256++` [.item-kind]#external#
+
+Returns the remaining number of tokens that `spender` will be allowed to spend on behalf of `owner` through <<transfer_from,transfer_from>>. This is zero by default.
+
+This value changes when <<approve,approve>> or <<transfer_from,transfer_from>> are called.
+
+[.contract-item]
+[[IERC20-transfer]]
+==== `[.contract-item-name]#++transfer++#++(recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Moves `amount` tokens from the caller's token balance to `to`.
+Returns `true` on success.
+
+Emits a <<Transfer,Transfer>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `recipient` is the zero address.
+
+[.contract-item]
+[[IERC20-transfer_from]]
+==== `[.contract-item-name]#++transfer_from++#++(sender: ContractAddress, recipient: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Moves `amount` tokens from `sender` to `recipient` using the allowance mechanism.
+`amount` is then deducted from the caller's allowance.
+Returns `true` on success.
+
+Emits a <<Transfer,Transfer>> event.
+
+This function panics if:
+
+- `sender` is the zero address.
+- `recipient` is the zero address.
+
+[.contract-item]
+[[IERC20-approve]]
+==== `[.contract-item-name]#++approve++#++(spender: ContractAddress, amount: u256) → bool++` [.item-kind]#external#
+
+Sets `amount` as the allowance of `spender` over the caller's tokens.
+Returns `true` on success.
+
+Emits an <<Approval,Approval>> event.
+
+This function panics if:
+
+- The caller is the zero address.
+- `spender` is the zero address.
+
+[.contract-item]
+[[IERC20-Transfer]]
+==== `[.contract-item-name]#++Transfer++#++(from: ContractAddress, to: ContractAddress, value: u256)++` [.item-kind]#event#
+
+Emitted when `value` tokens are moved from one address (`from`) to another (`to`).
+
+Note that `value` may be zero.
+
+[.contract-item]
+[[IERC20-Approval]]
+==== `[.contract-item-name]#++Approval++#++(owner: ContractAddress, spender: ContractAddress, value: u256)++` [.item-kind]#event#
+
+Emitted when the allowance of a `spender` for an `owner` is set by a call to <<approve,approve>>.
```

Comment:
```suggestion
Emitted when the allowance of a `spender` for an `owner` is set.
```
usually we don't control when an Event is emitted. Users might use internal methods in many ways. For example this event can be emitted through `increase_allowance`, not just `approve`.